packages/hydrogen/src/UIMsgButton.tsx

```diff
@@ -366,92 +342,80 @@ export const UIMsgButton = ({
-                        <Button.Title
-                            titleColor={contentColor}
-                            titleRole={UILabelRoles.ActionLabel}
-                            titleAnimStyle={titleAnim?.style}
-                        >
-                            {caption}
-                        </Button.Title>
-                    </Button.Content>
-                ),
+                <Button.Content direction={Button.ContentDirection.Column}>
+                    <Button.Title titleColor={contentColor} titleAnimStyle={titleAnim?.style}>
+                        {title}
+                    </Button.Title>
+                    <Button.Title
+                        titleColor={contentColor}
+                        titleRole={UILabelRoles.ActionLabel}
+                        titleAnimStyle={titleAnim?.style}
+                    >
+                        {caption}
+                    </Button.Title>
+                </Button.Content>,
                 null,
                 styles.doubleLineContent,
             ];
         }
 
         return [
-            (
-                <>
-                    <Button.Content>
-                        {
-                            iconPosition === UIMsgButtonIconPosition.Left && icon != null &&
-                            <Button.Icon
-                                source={icon}
-                                style={styles.leftIcon}
-                                iconAnimStyle={iconAnim?.style}
-                                initialColor={iconAnim?.initialColor}
-                                activeColor={iconAnim?.activeColor}
-                            />
-                        }
-                        {
-                            title != null &&
-                            <Button.Title
-                                titleColor={contentColor}
-                                titleAnimStyle={titleAnim?.style}
-                            >
-                                {title}
-                            </Button.Title>
-                        }
-                        {
-                            iconPosition === UIMsgButtonIconPosition.Middle && icon != null  &&
-                            <Button.Icon
-                                source={icon}
-                                iconAnimStyle={iconAnim?.style}
-                                initialColor={iconAnim?.initialColor}
-                                activeColor={iconAnim?.activeColor}
-                            />
-                        }
-                    </Button.Content>
-                    {
-                        iconPosition === UIMsgButtonIconPosition.Right && icon != null &&
+            <>
+                <Button.Content>
+                    {iconPosition === UIMsgButtonIconPosition.Left && icon != null && (
                         <Button.Icon
                             source={icon}
+                            style={!!title ? styles.leftIcon : null}
```

Comment:
So this is the fix, right? :)
Maybe extract to a variable and use this check instead in all places? (I like this check more than on 388 line) :)